common/scripts/setup_env.sh

```diff
@@ -103,12 +104,40 @@ if [[ -d "${HOME}/.config/gcloud" ]]; then
   CONDITIONAL_HOST_MOUNTS+="--mount type=bind,source=${HOME}/.config/gcloud,destination=/config/.config/gcloud,readonly "
 fi
 
-# Conditional host mount if KUBECONFIG is set
-if [[ -n "${KUBECONFIG}" ]]; then
-  CONDITIONAL_HOST_MOUNTS+="--mount type=bind,source=$(dirname "${KUBECONFIG}"),destination=/home/.kube,readonly "
-elif [[ -f "${HOME}/.kube/config" ]]; then
-  # otherwise execute a conditional host mount if $HOME/.kube/config is set
-  CONDITIONAL_HOST_MOUNTS+="--mount type=bind,source=${HOME}/.kube,destination=/home/.kube,readonly "
+# This function is designed for maximum compatibility with various platforms. This runs on
+# any Mac or Linux platform with bash 4.2+. Please take care not to modify this function
+# without testing properly.
+#
+# This function will properly handle any type of path including those with spaces using the
+# loading pattern specified by kubectl config.
+#
+# testcase: "a:b c:d"
+# testcase: "a b:c d:e f"
+# testcase: "a b:c:d e"
+parse_KUBECONFIG () {
+if [[ "$1" =~ ([^:]*):(.*) ]]; then
+  while true; do
+    rematch=${BASH_REMATCH[1]}
+    container_kubeconfig+="/home/$(basename "${rematch}"):"
```

Comment:
Remove the $(base name). It causes the KUBECONFIG entries to be too short.  It should be like: `KUBECONFIG=/home//Users/ericvn/work/go/src/istio.io/istio/a/.kube/config:/home//Users/ericvn/work/go/src/istio.io/istio/b/.kube/config:/config/config` and not `KUBECONFIG=/home/config:/home/config:/config/config`

Reply:
I leave it as an exercise to remove the double // after home.

Reply:
If you also remove the / after home, it does seem to fix the extra // in the path for my testing.

Reply:
i changed all of this logic, worth a re-review.